docs/book/src/reference/envtest.md

```diff
@@ -96,13 +96,15 @@ You can use environment variables and/or flags to specify the `kubectl`,`api-ser
+| Variable name                                                             | Type | When to use                                                                                                                                                                                                                                                          |
+|---------------------------------------------------------------------------| :--- |:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `USE_EXISTING_CLUSTER`                                                    | boolean | Instead of setting up a local control plane, point to the control plane of an existing cluster.                                                                                                                                                                      |
+| `KUBEBUILDER_ASSETS`                                                      | path to directory | Point integration tests to a directory containing all binaries (api-server, etcd and kubectl).                                                                                                                                                                       |
+| `TEST_ASSET_KUBE_APISERVER`, `TEST_ASSET_ETCD`, `TEST_ASSET_KUBECTL`      | paths to, respectively, api-server, etcd and kubectl binaries | Similar to `KUBEBUILDER_ASSETS`, but more granular. Point integration tests to use binaries other than the default ones. These environment variables can also be used to ensure specific tests run with expected versions of these binaries.                         |
+| `KUBEBUILDER_CONTROLPLANE_START_TIMEOUT` and `KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT` | durations in format supported by [`time.ParseDuration`](https://golang.org/pkg/time/#ParseDuration) | Specify timeouts different from the default for the test control plane to (respectively) start and stop; any test run that exceeds them will fail.                                                                                                                   |
+| `KUBEBUILDER_ATTACH_CONTROL_PLANE_OUTPUT`                                 | boolean | Set to `true` to attach the control plane's stdout and stderr to os.Stdout and os.Stderr. This can be useful when debugging test failures, as output will include output from the control plane.                                                                     |
+ | `TEST_ASSET_CERT_MANAGER`                                                 | path to directory | Point integration tests to a directory containing cert-manager binaries. Follow [cert-manager documentation](https://cert-manager.io/docs/installation/) to install it.                                                                                              |
```

Comment:
@yashsingh74, 

Did you test these options `TEST_ASSET_CERT_MANAGER` and `TEST_ASSET_PROMETHEUS`? I cannot find these ones in the controller-runtime implementation. Where do you check that env vars can be used to set it for the env test?

Reply:
Also, did you check the envtest config options for webhooks https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest#WebhookInstallOptions? Did you try to use them to config the cert and etc?

Reply:
Hi @yashsingh74,  

By looking at the issue https://github.com/kubernetes-sigs/kubebuilder/issues/2777 (it was opened too long ago and it is might out of context nowadays), I think that we can add here:

See that the envtest is commonly used to test the controllers.It is an integration test to "unit" test the controller/reconcile implementation. 

Therefore, we do not need use the metrics or cert-manager to run these tests usually. For example, look: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go

However, if you want to run the tests against a cluster where you have some configurations in place they can use the option to test using an existing cluster:

```golang
                testEnv = &envtest.Environment{
			UseExistingCluster: true,
		}
```

Note that we can link/share the ENV config options: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest#Environment

Also, we can let the users know that by using the plugin deploy-image/v1-alpha they can check an example of the code generated to use env tests to test the controller implementation via a note. See:
- https://book.kubebuilder.io/quick-start.html#using-deploy-image-plugin-to-generate-apis-and-controllers-code
- https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go

On top of that, share that if they are using webhooks they might want to check the optional config options https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest#WebhookInstallOptions